pal03377/Athena-New
# More usable submission detail for evaluation mode

### Motivation and Context



QoL improvements for viewing and assessing a submission and its feedback for both text and programming exercises. 
This is in preparation of the human evaluation mode and will be used to annotate the results.

### Description


Improves submission and exercise detail for text and programming exercises by integrating the Monaco text/code editor.
Displays the feedback inline and make it easy to view, add, or edit feedback.

❗️We have to remove Feedback Editor Test before merging, you can use it for testing the editor's edit feature:
<img width="1329" alt="image" src="https://github.com/ls1intum/Athena/assets/5898705/513ba77e-d441-4ac6-9da4-50a61afbd615">

For now the feedback looks like this:
<img width="775" alt="image" src="https://github.com/ls1intum/Athena/assets/5898705/ff659738-1092-43b9-8417-82c6b59bff6a">

**TODO in follow-up PR:**
 - Automatic evaluation results row below the feedback
    - E.g. rating of non-functional requirements (helpfulness, correctness, verboseness, ...)
    - Human annotator can then 👎👍 or rate how much they agree with the automatic evaluation

### Steps for Testing


> If you have previously exported some data from Artemis there are some issues with it, i.e. feedback references are off by one. I fixed it in #64, just in case you want to test with the evaluation data.

1. Use the Feedback Editor Test to see feedback editing in action
    1.1 Test text exercises
    1.2 Test programming exercises
2. Check submission details and exercise details if they look correct (you can't edit feedback there, inline feedback is shown in `send feedback` and `receive feedback suggestions`)

### Screenshots


*Programming Feedback (Editor)*
<img width="1249" alt="Screenshot 2023-07-30 at 11 09 24" src="https://github.com/ls1intum/Athena/assets/5898705/5fc3c6bb-5341-4afd-ab28-0ffd8f8fdc50">

*Text Feedback (Editor)*
<img width="1279" alt="image" src="https://github.com/ls1intum/Athena/assets/5898705/27f199ed-c484-4def-84b7-2b1e91df56f7">
